notify/email/email.go

```diff
@@ -385,3 +398,18 @@ func (n *Email) getPassword() (string, error) {
 	}
 	return string(n.conf.AuthPassword), nil
 }
+
+func generateThreadRootID(alerts []*types.Alert, hostname string) string {
+	if len(alerts) == 0 {
+		return ""
+	}
+
+	// Use first alert as representative of the alert group.
+	alert := alerts[0]
```

Comment:
Doesn't that mean even if we have 30 alerts, whatever the first one is, that's what will be used for fingerprinting? What if that alert resolves, but the other 29 remain?